src/snowflake/snowpark/modin/plugin/_internal/join_utils.py

```diff
@@ -216,8 +265,8 @@ def assert_snowpark_pandas_types_match() -> None:
         left,
         right,
         how,
-        left_on,
-        right_on,
+        left_join_key,
```

Comment:
do we also need to update JoinOrAlignOrderedDataframeResultHelper construction also, seems we might missed that part before